### PR TITLE
Feature/accept offer from chat room

### PR DESCRIPTION
**Pull request by DavidSurkov**

### CODE REVIEW

File: src/pages/Chat/hooks/useChatData.ts [hunk 1]
```diff
@@ -43,12 +44,20 @@ const useChatData = () => {
     };
   };
 
+  const defineLastMessage = (room: IRoom): string => {
+    if (room.lastMessage.messageType === MessageTypeEnum.FROM_USER) {
+      return room.lastMessage.text;
+    }
+    return "System message";
```
Comment (milkavladislav) [inline src/pages/Chat/hooks/useChatData.ts]: send to localization

File: src/pages/Chat/components/Message/Message.tsx [hunk 2]
```diff
@@ -21,25 +30,48 @@ function Message({
...
+  offerStatus,
 }: IMessage) {
   const { t } = useTranslation();
 
+  const defineMessageTitle = (): string => {
+    if (messageType === MessageTypeEnum.FROM_SYSTEM) {
+      return t("Chat.systemMessage");
+    }
+    if (messageType === MessageTypeEnum.FROM_SYSTEM_OFFER) {
+      return "Offer";
```
Comment (milkavladislav) [inline src/pages/Chat/components/Message/Message.tsx]: also localization

File: src/pages/Chat/components/ChatRoom/ChatRoom.tsx [hunk 3]
```diff
@@ -62,6 +76,73 @@ function ChatRoom({ userSelfId, userRole }: IChatRoom) {
 
   const ref = useChatScroll(messages);
 
+  const isClient = userRole === UserRoleEnum.CLIENT;
+
+  const renderSendOfferButton = (): JSX.Element => {
```
Comment (milkavladislav) [inline src/pages/Chat/components/ChatRoom/ChatRoom.tsx:81]: IMHO, if you use switch here this part of code will shorter. And DRY

**Review by milkavladislav — APPROVED** [review]

**Review by milkavladislav — APPROVED** [review]

**Review by milkavladislav — APPROVED** [review]

File: src/pages/Chat/hooks/useChatData.ts (hunk 1 above)
Comment (DavidSurkov) [reply src/pages/Chat/hooks/useChatData.ts]: fixed

File: src/pages/Chat/components/Message/Message.tsx (hunk 2 above)
Comment (DavidSurkov) [reply src/pages/Chat/components/Message/Message.tsx]: done

File: src/pages/Chat/components/ChatRoom/ChatRoom.tsx (hunk 3 above)
Comment (DavidSurkov) [reply src/pages/Chat/components/ChatRoom/ChatRoom.tsx:81]: fixed

**Review by milkavladislav — APPROVED** [review]

**Pull request merged**